### PR TITLE
Module 1 Soccer Updates

### PR DESCRIPTION
**Pull request by anthonypetersen**
This PR addresses the following Issue:
* https://github.com/episphere/connect/issues/858
-----
## Background Details
* Additional questions have been added to Module 1 ( `WORK3B` and `WORK7B`). The responses in these questions need to be combined with the responses from questions `WORK3` and `WORK7` when fetching information from SOCcer.
-----
## Technical Changes
* made `data` and `modules` global variables to be used across functions in `questionnaire.js`
* modified dev, stage, and prod Soccer URLs in `config` folders
* updated `soccerFunction()` to `externalListeners()`
* added `buildSoccerResults()` to handle calling Soccer
* added constants and event listeners for `WORK3B` and `WORK7B`
* cleaned up code

### CODE REVIEW

**Comment by ramavavi** [comment]
I have observed a really good way how code dynamically generates the soccerURL based on the current location's host how it allows the function to adapt to different environments.

**Review by ramavavi — APPROVED** [review]

File: js/pages/questionnaire.js [hunk 1]
```diff
@@ -203,46 +202,64 @@ async function startModule(data, modules, moduleId, questDiv) {
...
+
+    let module1 = modules[fieldMapping.Module1.conceptId];
+
+    let title3 = module1['D_627122657'] ?? '';
+    let task3 = module1['D_796828094'] ?? '';
+    let title7 = module1['D_118061122'] ?? '';
+    let task7 = module1['D_518387017'] ?? '';
+    
     if (work3){
         work3.addEventListener("submit", async (e) => {
```
Comment (we-ai) [inline js/pages/questionnaire.js:226]: `async` not needed.

File: js/pages/questionnaire.js [hunk 2]
```diff
@@ -203,46 +202,64 @@ async function startModule(data, modules, moduleId, questDiv) {
...
+            task3 = e.target[0].value;
+            const soccerResults = await buildSoccerResults(title3, task3);
+
+            buildHTML(soccerResults, occuptn1);
         });
     }
-    let work7 = document.getElementById("D_118061122");
+
     if (work7){
         work7.addEventListener("submit", async (e) => {
```
Comment (we-ai) [inline js/pages/questionnaire.js:245]: `async` not needed.

File: js/pages/questionnaire.js [hunk 3]
```diff
@@ -347,6 +367,23 @@ export const blockParticipant = () => {
...
 
+const buildSoccerResults = async (title, task) => {
+
+    let soccerURL = '';
+
+    if(location.host === urls.prod) soccerURL = SOCcerProd;
+    else if(location.host === urls.stage) soccerURL = SOCcerStage;
+    else soccerURL = SOCcerDev;
+    
+    let soccerResults = await (await fetch(`${soccerURL}title=${title}&task=${task}&n=6`)).json();
```
Comment (we-ai) [inline js/pages/questionnaire.js:378]: Is error handling needed in for data fetching?

File: js/pages/questionnaire.js [hunk 4]
```diff
@@ -347,6 +367,23 @@ export const blockParticipant = () => {
 
 }
 
+const buildSoccerResults = async (title, task) => {
+
+    let soccerURL = '';
```
Comment (we-ai) [inline js/pages/questionnaire.js:372]: We can save one line by doing ` let soccerURL = SOCcerDev;`, and change its value for stage and prod cases.

**Review by we-ai — COMMENTED** [review]
Commented on minor things.

File: js/pages/questionnaire.js (hunk 3 above)
Comment (anthonypetersen) [reply js/pages/questionnaire.js:378]: Per Daniel, no.

**Pull request merged**